app/controllers/order_items_controller.rb

```diff
@@ -0,0 +1,50 @@
+class OrderItemsController < ApplicationController
+  before_action :current_user
+
+  def create
+    my_order
+    @order.save
+    cookies.signed[:order] = @order.id
+
+    item = @order.order_items.where(product_id: params[:product_id]).first
+
+    unless item.nil?
+      item.update(quantity: order_item_params[:quantity])
+    else
+      @order.order_items << item = OrderItem.create(order_item_params)
```

Comment:
This is a nitpick, but I recommend against doing more than one piece of logic on a single line.
Meaning: instead of appending the created OrderItem to the order_items collection in the same line that you create the OrderItem, making it something like:
```ruby
else
  item = OrderItem.create(order_item_params)
  @order.order_items << item
```
makes things clearer for those inheriting your code.
It is the Ruby Way to value explicitness over cleverness or line-shortening.